harmony-api/prisma/schema.prisma

```diff
@@ -0,0 +1,168 @@
+datasource db {
+  provider = "postgresql"
+  url      = env("POSTGRES_URL")
+}
+
+generator client {
+  provider = "prisma-client-js"
+}
+
+model User {
+  user_id    Int       @id @default(autoincrement())
+  username   String    @unique
+  password   String
+  created_on DateTime  @default(now())
+  active     Boolean?
+  posts      Post[]
+  likes      Like[]
+  comments   Comment[]
+  Artist     Artist[]
+  Follows    Follow[]  @relation("Following")
+  Followers  Follow[]  @relation("Follower")
+}
+
+model Artist {
+  artist_id    Int            @id @map("user_id")
+  artist_name  String         @unique
+  artist_alias String?
+  user         User           @relation(fields: [artist_id], references: [user_id])
+  albums       Album[]
+  songs        Song[]
+  Publish      Publish[]
+  RelateArtist RelateArtist[]
+}
+
+model Post {
+  post_id           Int            @id @default(autoincrement())
+  user_id           Int
+  title             String
+  post_time         DateTime       @default(now())
+  user              User           @relation(fields: [user_id], references: [user_id])
+  poll              Poll?          @relation(fields: [pollPost_id], references: [post_id])
+  discussion        Discussion?    @relation(fields: [discussionPost_id], references: [post_id])
+  review            Review?        @relation(fields: [reviewPost_id], references: [post_id])
+  likes             Like[]
+  comments          Comment[]
+  pollPost_id       Int?
+  discussionPost_id Int?
+  reviewPost_id     Int?
+  RelateTo          RelateTo[]
+  RelateSong        RelateSong[]
+  RelateAlbum       RelateAlbum[]
+  RelateArtist      RelateArtist[]
+}
+
+model Poll {
+  post_id Int          @id @map("post_id")
+  options PollOption[]
+  Post    Post[]
+}
+
+model PollOption {
+  post_id Int    @id
+  option  String
+  poll    Poll   @relation(fields: [post_id], references: [post_id])
+}
```

Comment:
Do we need to have these split into two tables if each set of poll options only corresponds to one post? Or could we just have:
```
model PollOption {
  pollOptionId Int @id @default(autoincrement()) @map("poll_option_id")
  postId Int @map("post_id")
  option  String
  post Post @relation(fields: [postId], references: [postId])
}
```
And then post can have 
```
model Post {
  ...
  pollOptions PollOption[]
}

Reply:
Since the whole inheritence thing doesnt really work in Prisma, I can see why this might work better considering the Poll model brought nothing new to the table. See (8ffa273)